src/main/java/org/spongepowered/api/service/error/ErrorReport.java

```diff
@@ -0,0 +1,147 @@
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.service.error;
+
+import org.spongepowered.api.command.CommandSource;
+
+import java.net.URL;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * An error report. This object is mutable.
+ */
+public interface ErrorReport {
+
+    /**
+     * Get the {@link Throwable} that was responsible for creating this error report, if any is present.
+     *
+     * @return The optional throwable for this error report
+     */
+    Optional<Throwable> getCause();
+
+    /**
+     * Check if this error is the result of a misconfiguration on the user's part, or due to an error in programming.
+     * This defaults to true only if the {@link #getCause()} is a {@link UserErrorException}.
+     *
+     * @return Whether or not this is a user error
+     */
+    boolean isUserError();
+
+    /**
+     * Set whether or not this is user error.
+     *
+     * @param isUserError Whether or not this is a user error
+     * @return this
+     */
+    ErrorReport setUserError(boolean isUserError);
+
+    /**
+     * Get a description of the base cause of this error.
+     *
+     * @return The description of this error's base cause
+     */
+    String getDescription();
+
+    /**
+     * Get a section with the given title, creating and appending to the end of the error report as necessary.
+     *
+     * @param title The title to give the section
+     * @return The correct section
+     */
+    Category addCategory(String title);
+
+    /**
+     * Append a {@link ErrorReportable} to this report. If this {@link ErrorReportable} has already been added, it will not be added again.
+     *
+     * @param reportable The object to append information from
+     * @return this
+     */
+    ErrorReport addReportable(ErrorReportable reportable);
+
+    /**
+     * Dispatch this error as a fatal error. Fatal means that the server will be stopped and a full crash report will be generated.
+     *
+     * An exception will be used to pass the result of this error up the call stack. This exception must be passed on.
+     */
+    void dispatchFatal() throws RuntimeException;
+
+    /**
+     * Dispatch this error to a command source. If this command source is not the console, the error report will additionally be mirrored to the
+     * console.
+     *
+     * @param source The source to send information about the error report to
+     */
+    void dispatchToCommandSource(CommandSource source);
```

Comment:
Perhaps either replace this with, or add a new method for, dispatching to a `MessageReceiver`